core/src/main/java/org/apache/accumulo/core/logging/ScanUserDataLogger.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.accumulo.core.logging;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.slf4j.Marker;
+import org.slf4j.event.Level;
+import org.slf4j.helpers.AbstractLogger;
+import org.slf4j.spi.LoggingEventBuilder;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+
+public class ScanUserDataLogger extends AbstractLogger {
```

Comment:
Alternate implementation?

```
public class ScanUserDataLogger {

  private static final Logger LOG = LoggerFactory.getLogger(ScanUserDataLogger.class);
  private static final String FORMAT = "(%s) ";

  /**
   * Utility method that will log the msg and args to the calling class' logger (classLogger)
   * if the classLogger argument is supplied. The classLogger argument would be null in the case
   * where we don't want to log the same thing twice.
   * 
   * @param level slf4j log level
   * @param classLogger calling class' slf4j logger
   * @param userData scan userData value
   * @param msg log message
   * @param args log message argument
   */
  public static void log(Level level, Logger classLogger, String userData, String msg, Object... args) {
    if (classLogger != null && classLogger.isEnabledForLevel(level)) {
      classLogger.atLevel(level).log(msg, args);
    }
    if (LOG.isEnabledForLevel(level)) {
      LOG.atLevel(level).log(String.format(FORMAT, userData), args);
    }
  }
```

Reply:
Elsewhere you had asked:

> Why are we passing null for the logger? Under what circumstances would we want to pass null vs. something else?

I was thinking that in most cases users would want to redirect this logging to a different log file to provide a more condensed view of scan related logs. Therefore, I opted to print an additional log message instead of updating the current log messages. So, in the cases where I'm adding a new log message, the classLogger argument would be null. In the case where I'm reusing an existing message, then classLogger is not null.

I'm open to ideas here.

Reply:
That's a lot simpler, but I don't think it's as elegant as what @keith-turner did for `core/src/main/java/org/apache/accumulo/core/logging/TabletLogger.java`. I think it would be better to model it after that.

Reply:
I just want to prefix any msg with the userData. I don't see how creating a new method for each and every msg I want to log makes sense.

Reply:
> I just want to prefix any msg with the userData. I don't see how creating a new method for each and every msg I want to log makes sense.

I think what @ctubbsii  suggested could accomplish your goal, but would be out of scope for this PR.  It would need to be its own PR, a PR to gather all logging related to the lifecycle (client and server)  of scans into one place in the code.  The static methods are not the goal in TabletLogger.java, but one way of achieving some goals.  That code has the following two goals.

 * Use logical names for logging related to functionality instead of class names.  For example could use `org.apache.accumulo.scan` for all scan related logging, even if that functionality exist across many classes. Then a user could set org.apache.accumulo.scan to TRACE and redirect it to its own file to gather very detailed trace logging related to scans without overwhelming the general debug logs.
 * Gather all the code related to logging for specific functionality that user may interested in to one place in the code.  Then a user who wants to parse and search log messages can look at this one package in Accumulo and see what log messages are available to parse.